### PR TITLE
Fetch

### PR DESCRIPTION
**Pull request by Hanttone**
I added a server and a database (with dummy data) for the app, the API has two service points the germannouns and scores. 
Both of these endpoints can do POST and GET (germannouns will in the future only have a GET function). Furthermore, I built in a fetch into the app and can successfully fetch the words and also get a correct or wrong answer based on the button clicks on der, die and das buttons. 

I am not sure if Vercel is still able to run this, contemplating to change hosting to Heroku so please bear with me in case you can't yet click through the app and not get errors. :) Thanks a lot for your time and looking forward to your comments!

### CODE REVIEW

File: client/src/Pages/Game.js [hunk 1]
```diff
@@ -9,7 +10,50 @@ import Scores from '../Components/Scores/Scores';
...
+  const [
+    displayAnswer,
+    setDisplayAnswer,
+  ] = useState(true);
+
+  function randomWord() {
+    const randomNumber = Math.floor(
+      Math.random() * wordData.length
+    );
+    const selectedWord =
```
Comment (luschnat-ziegler) [inline client/src/Pages/Game.js:31]: Maybe rename the const as not to mix it up with the state variable above.

File: client/src/Pages/Game.js [hunk 2]
```diff
@@ -9,7 +10,50 @@ import Scores from '../Components/Scores/Scores';
...
-export default function GamePage() {
+export default function GamePage({ wordData }) {
+  const [
+    selectedWord,
+    setSelectedWord,
+  ] = useState('');
+  const [
+    correctAnswer,
+    setCorrectAnswer,
+  ] = useState('');
```
Comment (luschnat-ziegler) [inline client/src/Pages/Game.js:21]: Maybe you could bundle selecteWord and correctAnswer into one state.

File: client/src/Pages/Game.js [hunk 3]
```diff
@@ -9,7 +10,50 @@ import Scores from '../Components/Scores/Scores';
...
+    );
+    if (wordData[index].gender === nounGender) {
+      console.log('you got 10 points');
+      randomWord();
+    } else {
+      const correctAnswer = nounGender;
+      setCorrectAnswer(correctAnswer);
+      setDisplayAnswer(false);
+      randomWord();
+      setTimeout(
```
Comment (luschnat-ziegler) [inline client/src/Pages/Game.js:50]: You should handle the timer asynchronous(with a promise and a state).

File: client/src/Services/fetchData.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    const arrayOfPromises = dataUrls.map((url) =>
+      fetch(url)
+    );
+    for await (let request of arrayOfPromises) {
+      const data = await request.json();
+      allData.push(data);
+    }
+    return allData;
+  } catch (error) {
+    console.log(error.message);
```
Comment (FabianAlleblas) [inline client/src/Services/fetchData.js:19]: At some point I would recommend a visual error handling, so the user knows why she/he can't play 😅

File: client/src/App.js [hunk 5]
```diff
@@ -5,12 +5,24 @@ import {
   Switch,
```
Comment (FabianAlleblas) [inline client/src/App.js:5]: You could import the BrowserRouter/Router to the index.js

**Review by FabianAlleblas — COMMENTED** [review]
Good job, just 2 little comments 😃

File: client/src/Pages/Game.js [hunk 6]
```diff
@@ -9,7 +10,50 @@ import Scores from '../Components/Scores/Scores';
...
+  ] = useState('');
+  const [
+    correctAnswer,
+    setCorrectAnswer,
+  ] = useState('');
+  const [
+    displayAnswer,
+    setDisplayAnswer,
+  ] = useState(true);
+
```
Comment (Hecktors) [inline client/src/Pages/Game.js:26]: `displayAnswer` sounds like a function name, but it seems to be a boolean. What about something like `isAnswerDisplayed`?

File: client/src/Pages/Game.js [hunk 7]
```diff
@@ -9,7 +10,50 @@ import Scores from '../Components/Scores/Scores';
...
+  }
+
+  useEffect(() => randomWord(), []);
+
+  function handleClick(nounGender) {
+    const index = wordData.findIndex(
+      (word) => word.germanNoun === selectedWord
+    );
+    if (wordData[index].gender === nounGender) {
+      console.log('you got 10 points');
```
Comment (Hecktors) [inline client/src/Pages/Game.js:43]: Is it part of the game to log it into the console?

**Review by Hecktors — COMMENTED** [review]
Great job. Just want to play your game! 😉

File: client/src/Pages/Game.js (hunk 6 above)
Comment (Hanttone) [reply client/src/Pages/Game.js:26]: Thanks for the suggestion! :)

File: client/src/Pages/Game.js (hunk 7 above)
Comment (Hanttone) [reply client/src/Pages/Game.js:43]: Thanks it was just there for me when coding, must have forgotten to delete it.

**Pull request merged**